frontend/src/bundles/auth/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,31 @@
+import { type FC, type ReactNode } from 'react';
+import { Navigate } from 'react-router-dom';
+
+import { Loader } from '~/bundles/common/components/loader/loader.js';
+import { AppRoute } from '~/bundles/common/enums/app-route.enum.js';
+import { DataStatus } from '~/bundles/common/enums/data-status.enum.js';
+import { useAppSelector } from '~/bundles/common/hooks/hooks.js';
+
+type ProtectedRouteProperties = {
+    children: ReactNode;
+};
+
+const ProtectedRoute: FC<ProtectedRouteProperties> = ({ children }) => {
+    const { dataStatus } = useAppSelector(({ users }) => ({
```

Comment:
looks like we should check the user. Data can fulfilled without the user
`const hasUser = Boolean(user)`

Reply:
```js
 //TODO: update when user data will be stored in the store
 const hasUser = Boolean(user);
```

Reply:
@nikita-remeslov Do I understand right that for now I should replace the check with const hasUser = false to make it constantly false, until we can get the user?   I made the check for dataStatus for now because I had seen that there is no user in store for now, and I thought that this check would be better, but I can change it to constant "false"

Reply:
@MChornomaz Yes that's right. Then, we will check the presence of the user in the store, if it is not there, we will redirect to the login page

Reply:
@nikita-remeslov  Ok. Thanks, I will change that right after meeting)